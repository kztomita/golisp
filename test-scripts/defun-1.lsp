(defun foo ()
  (defun bar ()
    (print "bar called"))
  (bar))

; (bar) ここではまだ存在しない
(foo)
(bar) ; 外からでも呼べる

; 上書き
(defun bar ()
  (print "overwited bar called"))

(bar)

; barはglobalに一つしか存在しない

