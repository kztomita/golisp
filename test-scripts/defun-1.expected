"bar called"
"bar called"
"overwited bar called"
